spec/vehicle_factory_spec.rb

```diff
@@ -0,0 +1,159 @@
+# typed: ignore
+
+require "rspec"
+require "./spec/spec_helper"
+require "./lib/vehicle_factory"
+
+RSpec.describe VehicleFactory do # rubocop:disable Metrics/BlockLength
+  before(:each) do
+    @vehicle_factory = VehicleFactory.new
+    @wa_ev_registrations = DmvDataService.new.wa_ev_registrations
+    @ny_registrations = DmvDataService.new.ny_vehicle_registrations
+  end
+
+  it "exists" do
+    expect(@vehicle_factory).to be_an_instance_of(VehicleFactory)
+  end
+
+  describe "#create vehicle array" do
+    it "can create vehicle info" do
+      vehicle = @wa_ev_registrations[0]
+      info = @vehicle_factory.create_vehicle_hash(vehicle, :ev)
+      [
+        info[:engine],
+        info[:make],
+        info[:model],
+        info[:registration_date],
+        info[:vin],
+        info[:year]
+      ].each { |specification| expect(specification.nil?).to eq(false) }
+    end
+    it "can create an array of vehicles" do
+      @vehicle_factory.create_vehicles(@wa_ev_registrations, :ev)
+      expect(@vehicle_factory.vehicles[0]).to be_an_instance_of(Vehicle)
+    end
+  end
+
+  describe "#vehicle registration data" do # rubocop:disable Metrics/BlockLength
+    it "initializes with empty data set" do
+      @vehicle_factory.vehicle_data.each_value do |value|
+        expect(value).to eq({})
+      end
+    end
+    it "creates hashes with consolidated vehicle data" do
+      @vehicle_factory.create_vehicles(@wa_ev_registrations, :ev)
+      vehicle = @vehicle_factory.vehicles[0]
+      data = @vehicle_factory.retrieve_vehicle_data(vehicle)
+      data.each_value do |datum|
+        expect(datum.nil?).to eq(false)
+      end
+    end
+    it "counts the number of vehicles for each make" do
+      @vehicle_factory.create_vehicles(@wa_ev_registrations, :ev)
+      @vehicle_factory.retrieve_vehicle_makes_and_models
+      makes = {}
+      @vehicle_factory.vehicle_data[:makes].each do |make, make_data|
+        makes[make] = 0
+        make_data.each_value do |model|
+          makes[make] += model[:total]
+        end
+      end
+      total_vehicles = 0
+      makes.each_value do |vehicle_count|
+        total_vehicles += vehicle_count
+      end
+      expect(total_vehicles).to eq(@vehicle_factory.vehicles.count)
+    end
```

Comment:
For this test lines 51-65, there's too much dynamic code which can causes testing issues with results being off because the code in the test itself is off. Then we'd need to test the test. We'll learn more ways next week of how we can get around this issue when we're using data we don't control, but typically in our tests, we want to be more explicit of what the return values should be.